src/asyncopenstackclient/auth.py

```diff
@@ -1,43 +1,95 @@
 import aiohttp
+import os
 from dateutil import parser
 from time import time
+from urllib.parse import urljoin
 
 
 class AuthModel:
 
+    def __init__(self):
+        self._auth_url = None
+        self._username = None
+        self._password = None
+        self._user_domain_name = None
+        self._project_domain_name = None
+        self._project_id = None
+        self._project_name = None
+        self._region_name = None
+
     async def authenticate(self):
         raise NotImplementedError
 
+    @property
+    def os_auth_url(self):
+        return self._auth_url or os.environ.get('OS_AUTH_URL')
+
+    @property
+    def os_username(self):
+        return self._username or os.environ.get('OS_USERNAME')
+
+    @property
+    def os_password(self):
```

Comment:
I'm not so sure that this thing should be accessible

Reply:
It's either available through `_auth_dict` or `os.environ`, as well

Reply:
done